### PR TITLE
Release notes for 2.24

### PR DESCRIPTION
**Pull request by roberth**
# Motivation

Tell everyone about all the improvements that went into this significant release.

I'm not super happy about the way `zzz-other.md` ended up. I'm considering whether to split that up into proper release notes.

# Context

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

**Comment by edolstra** [comment]
I'm not sure if `zzz-other.md` adds a lot of value. It looks very messy, is probably not very useful to most readers, and people who really want that level of detail can always run `git log`. We want people to read the release notes so we shouldn't overwhelm them with detail. In particular bug fixes generally don't need to be listed in the release notes.

Note that contributing release notes is now the responsibility of PR authors, where previously it was the responsibility of the release manager. I don't want to go back to the release manager having to wade through months of commit logs to figure out what happened. So if we think the current release notes are lacking, we should push PR authors more to write them.

File: doc/manual/rl-next/zzz-other.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+synopsis: Other changes
+---
+
+- [#9063](https://github.com/NixOS/nix/pull/9063): introduce `libnixflake` and move flakes-specific code there (Nix is internally composed of a set of libraries, and this better reflects the architecture wrt flakes)
+- [#10852](https://github.com/NixOS/nix/pull/10852): make Nix commands respond better to interruption - Ctrl+C in the terminal
+- [#10853](https://github.com/NixOS/nix/pull/10853): fix docs of `builtins.importNative`
```
Comment (fricklerhandwerk) [inline doc/manual/rl-next/zzz-other.md:7]: https://github.com/NixOS/nix/pull/10853/files

This is not what the PR is about

File: doc/manual/rl-next/10734-nix3-build-show-all-fod-errors-with-keep-going.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+synopsis: "nix3-build: show all FOD errors with `--keep-going`"
+prs: 10734
+---
+
+The [`nix build`](@docroot@/command-ref/new-cli/nix3-build.md) command has been updated to improve the behavior of the [`--keep-going`] flag. Now, when `--keep-going` is used, all hash-mismatch errors of failing fixed-output derivations (FODs) are displayed, similar to the behavior of `nix build`. This enhancement ensures that all relevant build errors are shown, making it easier for users to update multiple derivations at once or to diagnose and fix issues.
+
+Author: [**Jörg Thalheim (@Mic92)**](https://github.com/Mic92)
```
Comment (Mic92) [inline doc/manual/rl-next/10734-nix3-build-show-all-fod-errors-with-keep-going.md:8]:
```suggestion
Author: [**Jörg Thalheim (@Mic92)**](https://github.com/Mic92), [**Maximilian Bosch**](https://github.com/Ma27)
```

File: doc/manual/rl-next/10734-nix3-build-show-all-fod-errors-with-keep-going.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+synopsis: "nix3-build: show all FOD errors with `--keep-going`"
+prs: 10734
+---
+
+The [`nix build`](@docroot@/command-ref/new-cli/nix3-build.md) command has been updated to improve the behavior of the [`--keep-going`] flag. Now, when `--keep-going` is used, all hash-mismatch errors of failing fixed-output derivations (FODs) are displayed, similar to the behavior of `nix build`. This enhancement ensures that all relevant build errors are shown, making it easier for users to update multiple derivations at once or to diagnose and fix issues.
```
Comment (Aleksanaa) [inline doc/manual/rl-next/10734-nix3-build-show-all-fod-errors-with-keep-going.md:6]:
```suggestion
The [`nix build`](@docroot@/command-ref/new-cli/nix3-build.md) command has been updated to improve the behavior of the [`--keep-going`] flag. Now, when `--keep-going` is used, all hash-mismatch errors of failing fixed-output derivations (FODs) are displayed, similar to the behavior of `nix-build`. This enhancement ensures that all relevant build errors are shown, making it easier for users to update multiple derivations at once or to diagnose and fix issues.
```

I guess?

Comment (roberth) [reply doc/manual/rl-next/10734-nix3-build-show-all-fod-errors-with-keep-going.md:6]: Whoops, thanks!

```suggestion
The [`nix build`](@docroot@/command-ref/new-cli/nix3-build.md) command has been updated to improve the behavior of the [`--keep-going`] flag. Now, when `--keep-going` is used, all hash-mismatch errors of failing fixed-output derivations (FODs) are displayed, similar to the behavior for other build failures. This enhancement ensures that all relevant build errors are shown, making it easier for users to update multiple derivations at once or to diagnose and fix issues.
```

**Comment by roberth** [comment]
> I'm not sure if the "Other changes" in `zzz-other.md` adds a lot of value.

Removed.



> Note that contributing release notes is now the responsibility of PR authors, where previously it was the responsibility of the release manager.

True, but as a maintainer team we do have the final responsibility to communicate with our users and especially contributors, as well as enabling contributors to write their own notes.


> we should push PR authors more to write them.

:+1:

**Review by edolstra — APPROVED** [review]

**Pull request merged**